GPU/Software/Rasterizer.cpp

```diff
@@ -114,11 +114,11 @@ static inline u8 ClampFogDepth(float fogdepth) {
 
 static inline void GetTexelCoordinates(int level, float s, float t, int& out_u, int& out_v)
 {
-	int width = 1 << (gstate.texsize[level] & 0xf);
-	int height = 1 << ((gstate.texsize[level]>>8) & 0xf);
+	int width = gstate.getTextureWidth(level);
+	int height = gstate.getTextureHeight(level);
 
-	int u = (int)(s * width);
-	int v = (int)(t * height);
+	int u = (int)(s * width + 0.375f);
+	int v = (int)(t * height + 0.375f);
```

Comment:
Turns out this is wrong.  In fact, I think we are half a pixel off for bilinear (i.e. a 0,0 -> 1,1 box will apply 50% of texel 1,1 in the center, which is of course wrong.)

-[Unknown]